api/processors/actions/group/discard.py

```diff
@@ -0,0 +1,62 @@
+from django.db import transaction
+
+from api.models import Group, Teacher
+from api.models.auxil.status_setter import StatusSetter
+from api.models.choices.log_event_type import StudentLogEventType, TeacherLogEventType
+from api.models.choices.status import StudentProjectStatus, TeacherProjectStatus
+from api.processors.actions.group import GroupActionProcessor
+from api.processors.auxil.log_event_creator import GroupLogEventCreator
+
+
+class GroupDiscardProcessor(GroupActionProcessor):
+    def __init__(self, group: Group, reason: str):
+        self.reason = reason
+        super().__init__(group)
+
+    @transaction.atomic
+    def process(self) -> None:
+        self._set_statuses()
+        self._create_log_events()
+        self._delete()
+
+    def _create_log_events(self) -> None:
+        GroupLogEventCreator.create(
+            group=self.group,
+            student_log_event_type=StudentLogEventType.TENTATIVE_GROUP_DISCARDED,
+            teacher_log_event_type=TeacherLogEventType.TENTATIVE_GROUP_DISCARDED,
+            comment=self.reason,
+        )
+
+    def _delete(self) -> None:
+        self.group.delete()
```

Comment:
Do we really want to completely delete group? Not to set some status as "deleted", or some flag, etc. 

It's better for history of changes.

@lemontree210

Reply:
> Do we really want to completely delete group? Not to set some status as "deleted", or some flag, etc.
> 
> It's better for history of changes.
> 
> @lemontree210

I know it sounds reasonable, but with the algorithm we might be creating dozens of tentative groups and I'm not sure if all of them are needed. They will be so many that the admins may not be able to use that archive anyway. We discuss something similar in #209 I think

Reply:
Ok, let's keep it as is and delete them